### ISSUE TITLE
SPARQL - Correct result in JDBC driver, but not in ISQL / SPARQL HTTP endpoint

### ISSUE THREAD

**Issue by Tomas2D** [openlink/virtuoso-opensource#1027]
Running following query from ISQL console
```sparql
SPARQL define output:format "JSON"
define sql:signal-void-variables 1

SELECT DISTINCT ?floorCount
WHERE {
  SERVICE <https://ruian.linked.opendata.cz/sparql> {
     <https://ruian.linked.opendata.cz/zdroj/stavební-objekty/22088181> <https://ruian.linked.opendata.cz/slovník/početPodlaží> ?floorCount
  }
}
```

gives me an empty result.

However, running the same query via JDBC driver gives me the following output

```
{
   "head":{
      "link":[],
      "vars":[
         "floorCount"
      ]
   },
   "results":{
      "distinct":false,
      "ordered":true,
      "bindings":[
         {
            "floorCount":{
               "type":"typed-literal",
               "datatype":"http://www.w3.org/2001/XMLSchema#integer",
               "value":"1"
            }
         }
      ]
   }
}
```

The `floorCount` is suddenly 1 however running mentioned query on target HTTP SPARQL Endpoint https://linked.opendata.cz/sparql gives me an empty result. This can be also verified via running `DESCRIBE <https://ruian.linked.opendata.cz/zdroj/stavební-objekty/22088181>`.

Running mentioned query via local SPARQL HTTP Endpoint yields to no matches. 
But probably due to issue reported in #1026. 
Running the query from ISQL yields to same result - no matches.

**Comment by Tomas2D** [comment]
It looks like that the values are probably random, because running the query mentioned before just with predicate 
`<https://ruian.linked.opendata.cz/slovník/početBytů>` results in number `145` (JDBC driver), but that triple does not exists on a target triple store (checked via describe and running same query from ISQL).

**Comment by Tomas2D** [comment]
It looks like that the JDBC driver gives me a correct result, because the corresponding HTML representation https://vdp.cuzk.cz/vdp/ruian/stavebniobjekty/22088181 shows numbers received from JDBC.

![Snímek obrazovky 2022-04-09 v 23 42 42](https://user-images.githubusercontent.com/15633909/162592460-2aaf6fbf-77c8-4155-abb3-c0c2b00ca684.png)

Can you please give me some insights or is this all happening because of #1026 ?

**Comment by HughWilliams** [comment]
By `ISQL console`, I presume you mean the Conductor "Database -> Interactive SQL" tab UI? which returns —

```pre
LONG VARCHAR
--
{ "head": { "link": [], "vars": [] }, "results": { "distinct": false, "ordered": true, "bindings": [ ] } }
No. of rows in result: 1
```

But, if the same query is run from the actual Virtuoso command line "isql" program, it returns —

```
SQL> SPARQL define output:format "JSON" define sql:signal-void-variables 1  SELECT DISTINCT ?floorCount WHERE {   SERVICE <https://ruian.linked.opendata.cz/sparql> {      <https://ruian.linked.opendata.cz/zdroj/stavební-objekty/22088181> <https://ruian.linked.opendata.cz/slovník/početPodlaží> ?floorCount   } };
callret-0
LONG VARCHAR
_______________________________________________________________________________


{ "head": { "link": [], "vars": ["floorCount"] },
  "results": { "distinct": false, "ordered": true, "bindings": [
    { "floorCount": { "type": "typed-literal", "datatype": "http://www.w3.org/2001/XMLSchema#integer", "value": "1" }} ] } }

1 Rows. -- 407 msec.
SQL>
```

— which is the same as for your JDBC connection as was seen in https://github.com/openlink/virtuoso-opensource/issues/1026, and would be expected.

So it looks like a problem with the Conductor iSQL UI, similar to running via the SPARQL endpoint, and needs to be looked into, although the Conductor iSQL UI should be running the query as "dba" so I would expect it to run?

**Comment by Tomas2D** [comment]
Exactly as you said. Running the query from "Interactive SQL" yields to an empty result set.

**Comment by pkleef** [comment]
I rechecked this issue after applying the fix for #1026 and it looks like this issue is resolved.

The /sparql endpoint, the iSQL page from the Conductor and the iSQL command line now all return the same data:
```
{ "head": { "link": [], "vars": ["floorCount"] },   "results": { "distinct": false, "ordered": true, "bindings": [     { "floorCount": { "type": "typed-literal", "datatype": "http://www.w3.org/2001/XMLSchema#integer", "value": "1" }} ] } }
```

**Comment by Tomas2D** [comment]
Perfect, thank you!
